pontoon/base/models.py

```diff
@@ -1586,9 +1586,47 @@ class UserProfile(models.Model):
     username = models.SlugField(unique=True, blank=True, null=True)
     bio = models.TextField(max_length=160, blank=True, null=True)
 
-    # Translation settings
-    quality_checks = models.BooleanField(default=True)
-    force_suggestions = models.BooleanField(default=False)
+    # External accounts
+    bugzilla = models.EmailField("Bugzilla email address", blank=True, null=True)
+    matrix = models.CharField("GitHub username", max_length=255, blank=True, null=True)
+    github = models.CharField("Matrix username", max_length=255, blank=True, null=True)
+
+    # Visibility
+    class Visibility(models.TextChoices):
+        ALL = "All users", "All users"
+        TRANSLATORS = "Translators", "Users with translator rights"
+
+    class VisibilityLoggedIn(models.TextChoices):
+        LOGGED_IN = "Logged in users", "Logged in users"
+        TRANSLATORS = "Translators", "Users with translator rights"
+
+    visibility_email = models.CharField(
+        "Email address",
+        max_length=20,
+        default=VisibilityLoggedIn.TRANSLATORS,
+        choices=VisibilityLoggedIn.choices,
+    )
+
+    visibility_external_accounts = models.CharField(
+        "External accounts",
+        max_length=20,
+        default=Visibility.TRANSLATORS,
```

Comment:
@flodolo Why is the default value not set to all users here? I'm afraid many people will set the external accounts and forgot to make them visible.

Reply:
I originally thought it would be better to err on the side of privacy.

1. The users who would be interested in using this data is likely admins (I want to cc someone to an issue or bug). The only exception would be Matrix, which was added later, but an all-or-nothing approach seems easier to implement.
2. Users might sign up with an "anonymous" account on Pontoon, and not tie it publicly to other accounts that might reveal their identity (to avoid issues at their day job, or because translating that language might lead to problems in real life).

I guess for 2 they might simply not add their external accounts…

So, I'd be OK with having these public by default, but maybe have some sort of indicator that these fields are public (we didn't think of that in the specs).

Reply:
Two more thoughts:
* The Bugzilla username is actually an email address. So, it might be problematic to have that as "all users". Can it be set as `VisibilityLoggedIn.LOGGED_IN`?
* `max_length=20`. Is that the max-length of each of these accounts? GitHub has 39 characters, but the email on Bugzilla can be much longer.

Reply:
I guess there are arguments for both options, so let's stick to the spec. We can also make a change afterwards, although probably only for users that wouldn't have set the external accounts yet.

With the default being `TRANSLATORS`, we can also keep the Bugzilla visibility unchanged. Having a separate visibility setting for each external account would also require us to rethink the UX for visibility settings.

`max_length=20` applies to the visibility field value.